src/main/charts/bamboo/templates/_helpers.tpl

```diff
@@ -310,6 +310,11 @@ For each additional plugin declared, generate a volume mount that injects that l
 
 {{- define "bamboo.volumeClaimTemplates" -}}
 {{- if or .Values.volumes.localHome.persistentVolumeClaim.create .Values.bamboo.additionalVolumeClaimTemplates }}
+{{- if or .Values.volumes.localHome.persistentVolumeClaimRetentionPolicy.whenDeleted .Values.volumes.localHome.persistentVolumeClaimRetentionPolicy.whenScaled }}
+persistentVolumeClaimRetentionPolicy:
+    whenDeleted: {{.Values.volumes.localHome.persistentVolumeClaimRetentionPolicy.whenDeleted}}
```

Comment:
There will be a warning but StatefulSet started fine. 
```  
Warning  FailedUpdate      6m32s  statefulset-controller  update Pod jira-1 in StatefulSet jira failed error: Could not update claim local-home-jira-1 for delete policy ownerRefs:  │
│ Operation cannot be fulfilled on persistentvolumeclaims "local-home-jira-1": the object has been modified; please apply your changes to the latest version and try again
```

Maybe I'll change the condition to `and` instead of `or`.